examples/query_elasticsearch/elasticsearch_queuelength.py

```diff
@@ -0,0 +1,164 @@
+#
+# Usage             : To query a list of Service BC offices, run:
+#
+#                   : python3 elasticsearch_queuelength.py --config
+#                   :   officelist.txt
+#                   :   --username ES_USER --password ES_PASS
+#                   :   --endpoint ES_ENDPOINT --index ES_INDEX
+#
+#                   : where 'officelist.txt' is a text file containing the
+#                   : newline separated list of Service BC field offices you
+#                   : wish to query.
+
+
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import Search, Q
+import logging
+from argparse import ArgumentParser
+import json
+import sys
+import os
+import signal
+
+# Arguments parsing
+parser = ArgumentParser(
+  description='Counts queue length  for given Service BC field offices\
+               using Elastisearch and Snowplow.')
+parser.add_argument('-c', '--config', help='Config file listing the Service \
+                    BC field offies of interest.', required=True)
+parser.add_argument('-u', '--username', help='Username.', required=True)
+parser.add_argument('-p', '--password', help='Password.', required=True)
+parser.add_argument('-d', '--debug', help='Debug', action="store_true")
+parser.add_argument('-i', '--index', help='Index', required=True)
+parser.add_argument('-ep', '--endpoint', help='Endpoint', required=True)
+args = parser.parse_args()
+
+
+# Ctrl+C
+def signal_handler(signal, frame):
+    logger.debug('Ctrl+C pressed!')
+    sys.exit(0)
+
+
+# Ctrl+C handler
+signal.signal(signal.SIGINT, signal_handler)
+
+# Create stream handler for logging
+logger = logging.getLogger(__name__)
+formatter = logging.Formatter("%(asctime)s:[%(levelname)s]: %(message)s")
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+# Validate config file
+config = args.config
+if os.path.isfile(config):
+    with open(config) as f:
+        offices = tuple(line.rstrip() for line in f)
+else:
+    logger.error("a config file listing Sewrvice BC field offices by line is required. \
+                 Use --config <file_path>")
+
+# Assign credentials and collector information
+http_user = args.username
+http_pass = args.password
+endpoint = args.endpoint
+index = args.index
+
+client = Elasticsearch(endpoint, http_auth=(http_user, http_pass))
+logger.debug('Elastic search object: ', client)
+
+with open('serviceBCOfficeList.json') as json_file:
+    serviceCenters = json.load(json_file)
+
+for office in offices:
+    office_id = ''
+    for serviceCenter in serviceCenters:
+        if serviceCenter["cfms_poc.office_name"] == office:
+            office_id = serviceCenter['cfms_poc.office_id']
+
+    # Query for number of addcitizen events. Note: The derived timestamp looks
+    # weird but 'now/d+7h' is rounding down to the start of the day in UTC
+    # which is 1700 which is 17h00 local on the previous day, so I am adding
+    # 5 hours to get midnight.
```

Comment:
I you're adding 7 hours; not 5 hours. No need for opinion on syntax. I suggest something like:

```
The values of the derived timestamp parameters are in Elasticsearch API date math syntax.
'now/d+7h' rounds 'now' down to the nearest day: 00h00 UTC. The '+7h' adjusts that time to 00h00 PST.
reference: https://www.elastic.co/guide/en/elasticsearch/reference/6.3/common-options.html#date-math
```
Have you tried the `time_zone` range query parameter documented here: https://www.elastic.co/guide/en/elasticsearch/reference/6.3/query-dsl-range-query.html#_time_zone_in_range_queries ? Using that would be more explicit and simplify your date math for `gte`.

Reply:
Added time zone parameter to range query and removed unnecessary commenting